ext/-test-/file/newline_conv.c

```diff
@@ -0,0 +1,73 @@
+#include "ruby/ruby.h"
+#include "ruby/io.h"
+#include <fcntl.h>
+
+static VALUE
+open_with_rb_file_open(VALUE self, VALUE filename, VALUE read_or_write, VALUE binary_or_text)
+{
+    char fmode[3] = { 0 };
+    if (rb_eql(read_or_write, rb_id2sym(rb_intern("read")))) {
+        fmode[0] = 'r';
+    }
+    else if (rb_eql(read_or_write, rb_id2sym(rb_intern("write")))) {
+        fmode[0] = 'w';
+    }
+    else {
+        rb_raise(rb_eArgError, "read_or_write param must be :read or :write");
+    }
+
+    if (rb_eql(binary_or_text, rb_id2sym(rb_intern("binary")))) {
+        fmode[1] = 'b';
+    }
+    else if (rb_eql(binary_or_text, rb_id2sym(rb_intern("text")))) {
+
+    }
+    else {
+        rb_raise(rb_eArgError, "binary_or_text param must be :binary or :text");
+    }
+
+    return rb_file_open(StringValueCStr(filename), fmode);
+}
+
+static VALUE
+open_with_rb_io_fdopen(VALUE self, VALUE filename, VALUE read_or_write, VALUE binary_or_text)
+{
+    int omode = 0;
+    if (rb_eql(read_or_write, rb_id2sym(rb_intern("read")))) {
```

Comment:
There is `rb_sym_intern_ascii_cstr` that returns a `Symbol` from `char*`, and a `Symbol` can be compared by the value itself.
```suggestion
    if (read_or_write == rb_sym_intern_ascii_cstr("read")) {
```

Reply:
oh hey I didn't know this, thanks.

Reply:
Oh, `rb_sym_intern_ascii_cstr` isn't exported so I can't use it from an extension.... I wonder if it should be though? It seems pretty useful. I'll keep `rb_id2sym(rb_intern)` for now I suppose.